controllers/validator.go

```diff
@@ -48,11 +45,27 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 		return validVsl, err
 	}
 
+	// ENSURE UPGRADES --------------------------------------------------------
 	// check for VSM/Volsync DataMover (OADP 1.2 or below) syntax
 	if dpa.Spec.Features != nil && dpa.Spec.Features.DataMover != nil {
 		return false, errors.New("Delete vsm from spec.configuration.velero.defaultPlugins and dataMover object from spec.features. Use Velero Built-in Data Mover instead")
 	}
 
+	// check for ResticConfig (OADP 1.4 or below) syntax
+	if dpa.Spec.Configuration.Restic != nil {
+		return false, errors.New("Delete restic object from spec.configuration, use spec.configuration.nodeAgent instead")
+	}
+	// ENSURE UPGRADES --------------------------------------------------------
+
+	// DEPRECATIONS -----------------------------------------------------------
+	if dpa.Spec.Configuration.NodeAgent != nil && dpa.Spec.Configuration.NodeAgent.UploaderType == "restic" {
+		deprecationWarning := "(Deprecation Warning) Use kopia instead of restic in spec.configuration.nodeAgent.uploaderType, which is deprecated and will be removed in the future"
```

Comment:
add more info here, like restore will still be supported?

Reply:
Its fine, you do not need to treat errors/warning as a substitute for documentation. Support policy is best explained via docs IMHO